desc/objectives/_stability.py

```diff
@@ -367,3 +369,278 @@ def compute(self, params, constants=None):
             profiles=constants["profiles"],
         )
         return constants["transforms"]["grid"].compress(data["magnetic well"])
+
+
+class BallooningStability(_Objective):
+    """A type of ideal MHD instability.
+
+    Infinite-n ideal MHD ballooning modes are of significant interest.
+    These instabilities are also related to smaller-scale kinetic instabilities.
+    With this class, we optimize MHD equilibria against the ideal ballooning mode.
+
+    Targets the following metric:
+
+    f = w₀ sum(ReLU(γ-γ₀)) + w₁ max(ReLU(γ-γ₀))
```

Comment:
This is what @rahulgaur104 used in his most recent paper, though we could likely get a similar effect by just returning the growth rates and uses `bounds`? That would fit more closely with the other objectives, but not sure how much worse it might perform.

Reply:
I understand how it can be confusing because it is different. Can you explain how bounds work? Is it the same as I explained in Appendix D.1 of said paper?

Reply:
yeah, using `bounds=(-np.inf, gamma0)` would be the same as the sum term when it gets reduced down to sum of squares, the only thing missing would be the additional weight on the max growth rate.

Note also that the current one can be rewritten as 

`w.T @ ReLU(np.sort(gamma)-gamma0)` where `w = [w0+w1, w0,w0,w0,...]`

so if we really want to weight the max growth rate more we could sort the return values, so that if the user gives the regular `objective.weight` as an array they can get the same effect as this?

Reply:
hmm ok i tried it out and if we do it with bounds it increases the number of residuals from 1 to nalpha*nzeta0, and that seems to run out of memory even at low res, so we probably want to keep the reduction to a scalar inside the compute function.

That said, I found that in the tutorial example we get better results if `w1=0` (ie no additional weight on the max growth rate). Not sure how general it is, but maybe just the sum term is enough?

Reply:
Also, I did not notice that you replaced lambda with gamma. I'm going to revert this back lambda = gamma^2 to be consistent with COBRAVMEC.

Reply:
I'd prefer to not use lambda since we use that for several other things already

Reply:
OK, I'm a bit confused by notation I think. I was assuming the eigenfunction goes like $\exp(i\omega t)$, so then $d^2/dt^2 \rightarrow -\omega^2 = \gamma$?

Are you using $\gamma$ to be my $\omega$, and $\lambda$ to mean my $\gamma$?

Reply:
Yes, what you call gamma is my lambda (also lambda in  COBRAVMEC). And this lambda = -omega^2  = growth_rate^2. What I call gamma is the growth rate = sqrt(lambda) = \pm i omega.